various changes/fixes required for nocturne-ui changes

68p opened the pull request:
changes: 
- devices endpoint returns all devices now, instead of just connected
- added connect and disconnect endpoint for Bluetooth menu 
- add connect websocket message 
- keep websocket server alive on abrupt disconnect
- fixed `http: superfluous response.WriteHeader call` by setting header first

issues: 
- `write tcp [::1]:5000->[::1]:60338: write: broken pipe`, but the websocket still works so it doesn't seem to be that big of an issue

68p committed:
fix(devices): devices endpoint to get paired devices ...

... so it can be used for the bluetooth menu in nocturne-ui

68p committed:
feat: add connect endpoint

68p committed:
fix(websocket): keep alive after abrupt disconnect

68p committed:
feat: add disconnect endpoint

68p committed:
fix(connect): check supported profiles before connection

68p committed:
fix: add missed device connected payload

68p committed:
Revert "fix(connect): check supported profiles before connection"

This reverts commit 97b44c12d5c2f8163b85f7ab291b0c0e75d3b555.

68p committed:
chore: remove missed comments

68p committed:
fix: put http header first then response

68p committed:
feat: add connect websocket message

itsnebulalol commented on bluetooth/manager.go:
```diff
@@ -328,25 +337,113 @@ func (m *BluetoothManager) GetConnectedDevices() ([]utils.BluetoothDeviceInfo, e
+	log.Printf("Device connected: %s", devicePath)
+
+	if m.wsHub != nil {
+		m.wsHub.Broadcast(utils.WebSocketEvent{
```
I think the bluetooth/connect event should only be broadcasted for when its monitored (so it's not only broadcasted when the nocturne api endpoint is used, and can be broadcasted when the user triggers a connect from their phone).

68p replied:
fixed in https://github.com/usenocturne/nocturned/pull/5/commits/856ba9f2c4f88c23295a61ecf6885ccbbad05798

itsnebulalol commented on bluetooth/manager.go:
```diff
@@ -328,25 +337,113 @@ func (m *BluetoothManager) GetConnectedDevices() ([]utils.BluetoothDeviceInfo, e
+	}
+
+	if m.wsHub != nil {
+		m.wsHub.Broadcast(utils.WebSocketEvent{
```
Same as above (bluetooth/connect).

68p committed:
fix: sends 1 connect message instead of 2

- actually just changed the monitorDisconnects function to what it was originally
